lib/fetch-nearby.js

```diff
@@ -30,15 +30,25 @@ export default async function fetchNearby(request, region, assembly) {
       displayStart -= 1000;
       displayEnd += 1000;
     } else {
-      displayStart = genes[0].end - 2000;
-      displayEnd = genes[1].start + 2000;
+      if (genes.length > 1) {
+        displayStart = genes[0].end - 2000;
+        displayEnd = genes[1].start + 2000;
+      } else {
+        const regionStart = parseInt(region.split(":")[1].split("-")[0]);
+        if (regionStart < genes[0].start) {
+          displayStart = regionStart - 2000;
+          displayEnd = genes[0].end + 2000;
+        } else {
+          displayStart = genes[0].start - 2000;
+          displayEnd = regionStart + 2000;
+        }
+      }
```

Comment:
I feel like assigning the “1000” and “2000” to meaningful consts makes this a lot easier to maintain, and to discern their meaning.

Reply:
I will update accordingly. Thank you so much!